third_party/envoy-gateway/Makefile

```diff
@@ -26,7 +26,19 @@ init-source: $(ENVOY_GATEWAY_DOWNLOADED)
 $(ENVOY_GATEWAY_DOWNLOADED):
 	mkdir -p envoy-gateway
 	curl -sfL https://github.com/envoyproxy/gateway/archive/refs/tags/$(ENVOY_GATEWAY_VERSION).tar.gz | tar xz --strip-components 1 -C envoy-gateway
-	patch -d envoy-gateway -p1 < patches/0001-Bumping-x-net-and-containerd.patch
+
+# 	Apply patches for the specified Envoy Gateway version if patches directory exists.
+# 	This code checks for the presence of a patches directory corresponding to ENVOY_GATEWAY_VERSION.
+# 	If the directory exists, it iterates over all .patch files within it.
+# 	For each valid patch file, it applies the patch to the envoy-gateway directory using patch -p1.
+	if [ -d patches ]; then \
```

Comment:
Don't need this conditional because the .gitkeep will ensure that the directory always exists.